modules/puppetca/plugin_configuration.rb

```diff
@@ -1,13 +1,19 @@
 module ::Proxy::PuppetCa
   class PluginConfiguration
     def load_classes
-      require 'puppetca/puppetca_puppet_cert'
       require 'puppetca/dependency_injection'
       require 'puppetca/puppetca_api'
     end
 
-    def load_dependency_injection_wirings(container_instance, settings)
-      container_instance.dependency :puppet_cert, lambda { ::Proxy::PuppetCa::PuppetCert.new }
+    def load_programmable_settings(settings)
+      raise ::Proxy::Error::ConfigurationError, "Parameter ':puppet_version' is expected to have a non-empty value" if settings[:puppet_version].to_s.empty?
+
+      use_provider = settings[:use_provider]
+      use_provider = [use_provider].compact unless use_provider.is_a?(Array)
+      use_provider << (settings[:puppet_version].to_s >= '4.6' ? :puppetca_http_api : :puppetca_puppet_cert)
```

Comment:
Could you change this 6.0 per @juliantodt's suggestion? I'd comfortable merging this after that.

Reply:
@ekohl: Done.